org.adempiere.base/src/org/compiere/model/Query.java

```diff
@@ -856,6 +856,24 @@ private final String buildSQL(StringBuilder selectClause, boolean useOrderByClau
 			boolean isFullyQualified = !joinClauseList.isEmpty();
 			if (selectColumns != null && selectColumns.length > 0)
 			{
+				if (virtualColumns != null) {
+					for (String vc : virtualColumns) {
+						boolean found = false;
+						for (String sc : selectColumns) {
+							if (vc.equalsIgnoreCase(sc)) {
+								found = true;
+								break;
+							}
+						}
+						if (!found) {
+							// add vc to array selectColumns
+							String[] tmp = new String[selectColumns.length + 1];
+							System.arraycopy(selectColumns, 0, tmp, 0, selectColumns.length);
+							tmp[selectColumns.length] = vc;
+							selectColumns = tmp;
+						}
+					}
+				}

```

Comment:
isn't this should be a check on the order by columns instead of virtual columns ? If a table has column A, B and C, selectColumns A, B and Order By C, would it work ?

Reply:
> isn't this should be a check on the order by columns instead of virtual columns ? If a table has column A, B and C, selectColumns A, B and Order By C, would it work ?

Is the same as in postgres, you can order by columns not in the select clause, but you cannot order by "virtual columns"

So, in postgres (and in REST) this is valid:
```sql
SELECT Name FROM C_Charge ORDER BY C_Charge_ID   /* Valid */
```

But the query of virtual columns must be something like this:
```sql
SELECT Name, UPPER(Description) as UpperDescription FROM C_Charge ORDER BY UpperDescription   /* Valid */
```

But, because the virtual column is not in the select columns the query produced is this:
```sql
SELECT Name FROM C_Charge ORDER BY UpperDescription   /* Error */
```